### PR TITLE
2017 Cult Overhaul P1:  Cult Master and Datum Cult 

### PR DESCRIPTION
**Pull request by Robustin**
See: https://github.com/tgstation/tgstation/pull/23866 for prior details

Several miscellaneous fixes like Shade Mind's never getting removed from the cult member list (even when made into constructs) and a few of cult mode's checks simplified into global vars. 

- [x] Make the damn blood mark image show up on the target
- [x] Give construct's a "locate master" action button that will track their creator similar to the blood mark
- [x] Change the tracking indicator to something easier to see/understand at a glance
- [x] Bonus: Add functioning datum cult so cult doesn't utterly break outside of cult mode

🆑 Robustin
add: Blood Cultists can now attempt to claim the position of Cult Master with the approval of a majority of their brethren.
add: The Cult Master has access to unique blood magic that will aid them in leading the cult to victory:
add: The Cult Master can mark targets, allowing the entire cult to track them down.
add: The Cult Master has single-use, noisy, and overt channeled spell that will summon the entire cult to their location.
add: All cultists gain a HUD alert that will assist them in completing their objectives.
add: Constructs created by soul shards will now be able to track their master.
add: Cultists created outside of cult mode will now get a working sacrifice and summon objective and the summon rune will no longer fail outside of cult mode. 
/🆑

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
Add [WIP] prefixing the title 😄

**Comment by optimumtact** [comment]
this should just be given to someone at roundstart, attempting to make it a pick is just going to result in half the cultists ignoring the option chosen if they picked someone else

File: code/game/gamemodes/cult/cult.dm [hunk 1]
```diff
@@ -36,10 +34,10 @@
...
 	restricted_jobs = list("Chaplain","AI", "Cyborg", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel")
 	protected_jobs = list()
-	required_players = 24
-	required_enemies = 4
-	recommended_enemies = 4
-	enemy_minimum_age = 14
+	required_players = 1
+	required_enemies = 1
+	recommended_enemies = 1
+	enemy_minimum_age = 1
```
Comment (ChangelingRain) [inline code/game/gamemodes/cult/cult.dm]: debug~

Comment (Robustin) [inline code/game/gamemodes/cult/cult.dm]: wewps

File: code/game/gamemodes/cult/cult_comms.dm [hunk 2]
```diff
@@ -65,3 +69,144 @@
...
+	for(var/datum/mind/M in SSticker.mode.cult)
+		M.current << "<span class='cultlarge'><b>The blood mark on [GLOB.blood_target] has expired!</b></span>"
+	GLOB.blood_target = null
+
+
+obj/effect/cultmark
+	var/image/cult_marker
+	var/mob/living/viewer
+
+obj/effect/cultmark/Initialize(var/mob/living/T, var/atom/BT)
```
Comment (ChangelingRain) [inline code/game/gamemodes/cult/cult_comms.dm]: the first arg of Initialize must be `mapload`; any other args are passed from New on the assumption that the first arg of New is now the thing's `loc`

**Comment by Robustin** [comment]
@ExcessiveUseOfCobblestone 

Its all optional WIP stuff that will be done before this gets a serious review anyway. I want to push for test merging it so the core package is all done.

@optimumtact 

Absolutely no. The master has to self-nominate so there's no "slate of candidates" its just an up/down vote requiring a majority of the non-abstaining voters.

This role doesn't increase the individual's power, and leadership is generally not that popular of a position. See how many people are excited to be a revhead or nukeop's leader compared to the rank and file who just get to kill stuff and not worry about herding cats. 

We've never had a position in this game where your title is actually indicative of your leadership qualities. Being able to muster a majority vote (and there's a pre-vote period for discussion once someone is nominated and another roundstart window to prevent nomination rushes) gives at least some inkling that you aren't a random moron and thus will actually leverage your position for the good of the group - and in turn the position will actually deserve the respect that the title implies because the cult had some buy-in to the process as opposed to being forced to work under some randomly selected goober.

File: code/game/gamemodes/cult/cult_comms.dm (hunk 2 above)
Comment (Robustin) [inline code/game/gamemodes/cult/cult_comms.dm]: I actually had mapload in there and the first argument was the mark's turf - but removed it to see if thats what was preventing the image from appearing. That wasn't the case.

**Comment by optimumtact** [comment]
The only people who are going to bother listening to the leader is the people who voted for him so I don't see the point

**Comment by Robustin** [comment]
Worst case then twice as many people listen to someone calling the shots. 

Even if they don't listen, the marking system conveys useful information even if you think you dont care what your leader has to say.

**Comment by ExcessiveUseOfCobblestone** [comment]
I think newbies who don't know what they're doing are going to listen to the leader. Veterans who think they can do better may not, but they aren't really who this feature is aimed at if I've read robustin's posts correctly.

**Comment by Durkel** [comment]
The idea is that ANY direction is better than five people pulling at the same strings.By electing a leader democratically it should give a normally directionless team some form of direction.

File: code/__HELPERS/game.dm [hunk 3]
```diff
@@ -485,6 +485,59 @@
...
+		Nominee << "It would be premature to select a leader while everyone is still settling in, try again in [(1800-world.time)/10] seconds"
+		return
+	for(var/datum/mind/B in SSticker.mode.cult)
+		if(!istype(B.current, /mob/living))
+			continue
+		var/mob/living/M = B.current
+		if(isliving(M))
+			M << 'sound/misc/notice2.ogg' //Alerting them to their consideration
+			M.verbs -= /mob/living/proc/cult_master
+			M << "<span class='cultlarge'>[usr] has asserted that they are worthy of leading the cult. A vote will be called shortly.</span>"
```
Comment (swindly) [inline code/__HELPERS/game.dm]: I think you're supposed to use to_chat and p_they instead of << and they.

File: code/__HELPERS/game.dm [hunk 4]
```diff
@@ -485,6 +485,59 @@
...
+			spawn(200)
```
Comment (optimumtact) [inline code/__HELPERS/game.dm]: for what purpose

File: code/__HELPERS/game.dm [hunk 5]
```diff
@@ -485,6 +485,59 @@
...
+						M << 'sound/machines/buzz-sigh.ogg'
+					else
+						yes_voters += M
+						cult_total += M
+				if(2)
+					M << "<span class='danger'>Choice registered: No.</span>"
+					cult_total += M
+				if(3)
+					M << "<span class='danger'>Choice registered: Abstain.</span>"
+	sleep(400)
```
Comment (optimumtact) [inline code/__HELPERS/game.dm]: not great

File: code/__HELPERS/game.dm [hunk 6]
```diff
@@ -485,6 +485,59 @@
...
+	if(yes_voters.len > (cult_total.len/2))
+		var/datum/action/innate/cultmast/finalreck/FinalReckoning= new()
+		var/datum/action/innate/cultmast/cultmark/Mark= new()
+		FinalReckoning.Grant(Nominee)
+		Mark.Grant(Nominee)
+		Nominee.update_action_buttons_icon()
+		Nominee.mind.special_role = "Cult Master"
+		SSticker.mode.set_antag_hud(Nominee,"cultmaster")
+		GLOB.cult_mastered = 1
```
Comment (optimumtact) [inline code/__HELPERS/game.dm]: really, a global variable? just stick it in the SSticker

File: code/__HELPERS/game.dm [hunk 7]
```diff
@@ -485,6 +485,59 @@
...
+		for(var/datum/mind/B in SSticker.mode.cult)
+			if(isliving(B.current))
+				B.current << "<span class='cultlarge'>[Nominee] has the cult's support and is now their master. Follow their orders to the best of your ability!"
+		return TRUE
+	else
+		for(var/datum/mind/B in SSticker.mode.cult)
+			if(isliving(B.current))
+				B.current << "<span class='cultlarge'>[Nominee] could not win the cult's support and shall continue to serve as an acolyte. A new vote may be called shortly."
+				sleep(450)
```
Comment (optimumtact) [inline code/__HELPERS/game.dm]: without a spawn this is going to grant each person their verb back one by one separated by 450 seconds.

just use the add_timer and a callback method that adds the verb

File: code/_onclick/hud/alert.dm [hunk 8]
```diff
@@ -256,6 +258,74 @@ or shoot a gun to move around via Newton's 3rd Law of Motion."
...
+	var/datum/job/sacjob = SSjob.GetJob(sacmind.assigned_role)
+	var/datum/preferences/sacpref = sacmob.client.prefs
+	var/icon/reshape = get_flat_human_icon(null, sacjob, sacpref)
+	reshape.Shift(SOUTH, 4)
+	reshape.Crop(7,4,26,30)
+	reshape.Crop(-5,-2,26,29)
+	sacimage = reshape
+	add_overlay(finder)
+	START_PROCESSING(SSprocessing, src)
+	process()
```
Comment (optimumtact) [inline code/_onclick/hud/alert.dm]: why

File: code/_onclick/hud/alert.dm [hunk 9]
```diff
@@ -256,6 +258,74 @@ or shoot a gun to move around via Newton's 3rd Law of Motion."
...
+
+/obj/screen/alert/bloodsense/Destroy()
+    qdel(finder)
```
Comment (optimumtact) [inline code/_onclick/hud/alert.dm]: null sacimge and sacmob

use the QDEL_NULL macro on the finder

File: code/_onclick/hud/alert.dm [hunk 10]
```diff
@@ -256,6 +258,74 @@ or shoot a gun to move around via Newton's 3rd Law of Motion."
...
+		return
+	if(!GLOB.blood_target && GLOB.sac_complete)
+		cut_overlays()
+		finder.icon_state = "sacrifice"
+		desc = "The sacrifice is complete, prepare to summon Nar-Sie!"
+		add_overlay(finder)
+		return
+	var/turf/Q = get_turf(GLOB.blood_target)
+	var/turf/A = get_turf(mob_viewer)
+	if(Q.z != A.z) //The target is on a different Z level, we cannot sense that far.
```
Comment (optimumtact) [inline code/_onclick/hud/alert.dm]: what if it's not a turf, what if mob_veiwer doesn't exist or is null?

File: code/game/gamemodes/cult/cult.dm [hunk 11]
```diff
@@ -109,6 +108,7 @@
 					possible_targets += player.mind
 		if(possible_targets.len > 0)
 			sacrifice_target = pick(possible_targets)
+			GLOB.sac_target = sacrifice_target
```
Comment (optimumtact) [inline code/game/gamemodes/cult/cult.dm]: again why is this a global and not living on a subsystem ticker

File: code/game/gamemodes/cult/cult_comms.dm [hunk 12]
```diff
@@ -65,3 +69,144 @@
...
+	background_icon_state = "bg_demon"
+	buttontooltipstyle = "cult"
+	check_flags = AB_CHECK_RESTRAINED|AB_CHECK_STUNNED|AB_CHECK_CONSCIOUS
+
+/datum/action/innate/cultmast/finalreck/Activate()
+	var/list/destinations = list()
+	for(var/turf/T in orange(1,owner))
+		destinations += T
+	for(var/i in 1 to 4)
+		if(do_after(owner, 30, target = owner))
```
Comment (optimumtact) [inline code/game/gamemodes/cult/cult_comms.dm]: why does it recall everything four time

Comment (optimumtact) [inline code/game/gamemodes/cult/cult_comms.dm]: like you're setting the mob turf every single time

File: code/game/gamemodes/cult/cult_comms.dm [hunk 13]
```diff
@@ -65,3 +69,144 @@
...
+	var/time = 0
+
+/datum/action/innate/cultmast/cultmark/New()
+    CM = new()
+    ..()
+
+/datum/action/innate/cultmast/cultmark/IsAvailable()
+	if(owner.mind.special_role != "Cult Master")
+		return 0
+	if((world.time - time)<900)
```
Comment (optimumtact) [inline code/game/gamemodes/cult/cult_comms.dm]: use some defines onthese numbers please

File: code/game/gamemodes/cult/cult_comms.dm [hunk 14]
```diff
@@ -65,3 +69,144 @@
...
+		owner << "<span class='cultlarge'><b>You need to wait [round((900-(world.time-time))/10)] seconds before you can mark another target!</b></span>"
+		return 0
+	return ..()
+
+/datum/action/innate/cultmast/cultmark/Destroy()
+    qdel(CM)
```
Comment (optimumtact) [inline code/game/gamemodes/cult/cult_comms.dm]: QDEL_NULL macro

File: code/game/gamemodes/cult/cult_comms.dm [hunk 15]
```diff
@@ -65,3 +69,144 @@
...
+		for(var/datum/mind/M in SSticker.mode.cult)
+			M.current << "<span class='cultlarge'><b>Master [ranged_ability_user] has marked [GLOB.blood_target] as the cult's highest priority, get there immediately!</b></span>"
+			M.current << pick(sound('sound/hallucinations/over_here2.ogg',0,1,75), sound('sound/hallucinations/over_here3.ogg',0,1,75))
+			new /obj/effect/cultmark(M, GLOB.blood_target)
+		remove_ranged_ability(caller, "The marking rite is complete! It will last for one minute.")
+		addtimer(CALLBACK(GLOBAL_PROC, .proc/reset_blood_target), 1200, TIMER_OVERRIDE)
+		return TRUE
+	return FALSE
+
+/proc/reset_blood_target()
```
Comment (optimumtact) [inline code/game/gamemodes/cult/cult_comms.dm]: why is this a global proc

File: code/game/gamemodes/cult/cult_comms.dm [hunk 16]
```diff
@@ -65,3 +69,144 @@
...
+	var/image/cult_marker
+	var/mob/living/viewer
+
+obj/effect/cultmark/Initialize(var/mob/living/T, var/atom/BT)
+	..()
+	viewer = T
+	cult_marker = image('icons/effects/cult_target.dmi', BT, "glow", ABOVE_MOB_LAYER)
+	if(viewer.client)
+		viewer.client.images |= cult_marker
+	sleep(1200)
```
Comment (optimumtact) [inline code/game/gamemodes/cult/cult_comms.dm]: qdel_in macro instead of this sleep qdel please

File: code/game/gamemodes/cult/cult_comms.dm [hunk 17]
```diff
@@ -65,3 +69,144 @@
...
+	qdel(src)
+
+/obj/effect/cultmark/Destroy()
+	if(viewer.client)
+		viewer.client.images.Remove(cult_marker)
```
Comment (optimumtact) [inline code/game/gamemodes/cult/cult_comms.dm]: null the cult_marker image as well

File: code/game/gamemodes/cult/cult_items.dm [hunk 18]
```diff
@@ -213,7 +213,7 @@
 	flags_inv = HIDEJUMPSUIT
 	allowed = list(/obj/item/weapon/tome,/obj/item/weapon/melee/cultblade)
 	body_parts_covered = CHEST|GROIN|LEGS|ARMS
-	armor = list(melee = -50, bullet = -50, laser = -100,energy = -50, bomb = -50, bio = -50, rad = -50, fire = 0, acid = 0)
+	armor = list(melee = -50, bullet = -50, laser = -50,energy = -50, bomb = -50, bio = -50, rad = -50, fire = 0, acid = 0)
```
Comment (optimumtact) [inline code/game/gamemodes/cult/cult_items.dm]: why did you adjust the cult robe

File: code/modules/mob/living/death.dm [hunk 19]
```diff
@@ -49,8 +49,14 @@
 	timeofdeath = world.time
 	tod = worldtime2text()
 	var/turf/T = get_turf(src)
+	var/area/A = get_area(T)
+	if(src.mind.special_role == "Cult Master")
```
Comment (optimumtact) [inline code/modules/mob/living/death.dm]: I don't see a reason for this and it's awfully snowflake for a single roundtype

Comment (Robustin) [inline code/modules/mob/living/death.dm]: Well whats an alternative?

File: code/game/gamemodes/cult/cult_items.dm (hunk 18 above)
Comment (Robustin) [inline code/game/gamemodes/cult/cult_items.dm]: Should be obvious. I bumped the nerfed the damage increase from 100% to 50% last year and clearly missed a value.

File: code/game/gamemodes/cult/cult_comms.dm (hunk 16 above)
Comment (Robustin) [inline code/game/gamemodes/cult/cult_comms.dm]: I can't even get this to work, I have scarcely a clue on how client-side images work

File: code/game/gamemodes/cult/cult_comms.dm (hunk 12 above)
Comment (Robustin) [inline code/game/gamemodes/cult/cult_comms.dm]: Its like all the clock cult spells where the channeling is broken up into parts for effect.

File: code/game/gamemodes/cult/cult.dm (hunk 11 above)
Comment (Robustin) [inline code/game/gamemodes/cult/cult.dm]: I missed the tutorial on how to make subsystem tickers function like global variables

File: code/_onclick/hud/alert.dm (hunk 8 above)
Comment (Robustin) [inline code/_onclick/hud/alert.dm]: because I copied this from the clock cult alert

File: code/__HELPERS/game.dm (hunk 4 above)
Comment (Robustin) [inline code/__HELPERS/game.dm]: To have a brief discussion window before people are prompted to vote. Though this is going to end with you telling me to do some callback thingy.

File: code/game/gamemodes/cult/cult.dm (hunk 11 above)
Comment (ExcessiveUseOfCobblestone) [inline code/game/gamemodes/cult/cult.dm]: replace glob with SSticker

File: code/modules/mob/living/death.dm [hunk 20]
```diff
@@ -49,8 +49,14 @@
 	timeofdeath = world.time
 	tod = worldtime2text()
 	var/turf/T = get_turf(src)
+	var/area/A = get_area(T)
+	if(src.mind.special_role == "Cult Master")
+		for(var/datum/mind/M in SSticker.mode.cult)
+			if(istype(M.current, /mob/living) && isliving(M.current))
```
Comment (ChangelingRain) [inline code/modules/mob/living/death.dm]: these are the exact same check

File: code/modules/mob/living/death.dm [hunk 21]
```diff
@@ -49,8 +49,14 @@
 	timeofdeath = world.time
 	tod = worldtime2text()
 	var/turf/T = get_turf(src)
+	var/area/A = get_area(T)
+	if(src.mind.special_role == "Cult Master")
+		for(var/datum/mind/M in SSticker.mode.cult)
+			if(istype(M.current, /mob/living) && isliving(M.current))
+				M.current << "<span class='cultlarge'> Your cult's master, [mind.name], has fallen in the [A.name]!"
+			else
+				continue
```
Comment (ChangelingRain) [inline code/modules/mob/living/death.dm]: you don't need this else continue because it does that automatically

File: code/modules/mob/living/death.dm (hunk 19 above)
Comment (ChangelingRain) [inline code/modules/mob/living/death.dm]: A status effect given to the cult master that deletes itself when their stat is DEAD, and when it's deleted it does this stuff.

File: code/game/gamemodes/cult/cult_comms.dm (hunk 16 above)
Comment (optimumtact) [inline code/game/gamemodes/cult/cult_comms.dm]: QDEL_IN(thing, time)

File: code/game/gamemodes/cult/cult_comms.dm (hunk 12 above)
Comment (optimumtact) [inline code/game/gamemodes/cult/cult_comms.dm]: okay no problem there

File: code/game/gamemodes/cult/cult_items.dm (hunk 18 above)
Comment (optimumtact) [inline code/game/gamemodes/cult/cult_items.dm]: if it's a change, then put it in the changelog

Comment (optimumtact) [inline code/game/gamemodes/cult/cult_items.dm]: and mention it in your OP

Comment (optimumtact) [inline code/game/gamemodes/cult/cult_items.dm]: it should be in a seperate PR though if if it is abalance change since it has no relationship with the rest of these changes

File: code/_onclick/hud/alert.dm (hunk 8 above)
Comment (optimumtact) [inline code/_onclick/hud/alert.dm]: You shouldn't need to process once here, just leave it for the ticker to process

File: code/__HELPERS/game.dm (hunk 4 above)
Comment (optimumtact) [inline code/__HELPERS/game.dm]: a spawn only splits out things that are indented beneath it and sleeps for the time given and then triggers that code.

Since nothing is indented this spawn does nothing, you probably wanted a sleep

File: code/modules/mob/living/death.dm (hunk 19 above)
Comment (Robustin) [inline code/modules/mob/living/death.dm]: Is there example I can use? I don't know of any status effect besides diseases/mutations that are permanent and disable on death.

File: code/_onclick/hud/alert.dm (hunk 10 above)
Comment (Robustin) [inline code/_onclick/hud/alert.dm]: I could add a check for the latter but if it exists wouldn't it always have a turf?

File: code/game/gamemodes/cult/cult.dm (hunk 11 above)
Comment (Robustin) [inline code/game/gamemodes/cult/cult.dm]: Calling it isn't the hard part, I dont know how to add a var to a specific game ticker. I'll figure it out I guess.

File: code/modules/mob/living/death.dm (hunk 21 above)
Comment (Robustin) [inline code/modules/mob/living/death.dm]: I made this PR shortly after I returned and my coding was even more atrocious than it is now.

File: code/game/gamemodes/cult/cult_items.dm (hunk 18 above)
Comment (Robustin) [inline code/game/gamemodes/cult/cult_items.dm]: Its literally a tiny bug fix. all the values should be -50, one was missed.

**Comment by ChangelingRain** [comment]
I mean status effects as in `/datum/status_effect`; you'd just check stat in tick(), deleting the effect if the user's stat is DEAD and do the message in on_remove()

**Comment by Robustin** [comment]
That feels just as snowflakey and more expensive than a check that occurs on mob death, but whatever.

**Comment by ChangelingRain** [comment]
It's more expensive but the system is less snowflakey because it's self-contained.

File: code/__HELPERS/game.dm (hunk 4 above)
Comment (Robustin) [inline code/__HELPERS/game.dm]: I know, I coded this shortly after I returned and have learned much since then. But I will say that spawn does work like this. They delay is there and working as intended.

Comment (optimumtact) [inline code/__HELPERS/game.dm]: it's confusing, make it a sleep

File: code/game/gamemodes/cult/cult_comms.dm (hunk 17 above)
Comment (optimumtact) [inline code/game/gamemodes/cult/cult_comms.dm]: cult_marker = null

File: code/game/gamemodes/cult/cult_items.dm (hunk 18 above)
Comment (optimumtact) [inline code/game/gamemodes/cult/cult_items.dm]: it's a balance change even if it is a tiny bugfix

**Comment by Robustin** [comment]
Jesus christ why is it so hard to get a client-side image to show up.

**Comment by Robustin** [comment]
God damnit of course lzimann would refactor antag datums while I'm shifting the entire mode to rely on them more.

File: code/__HELPERS/game.dm [hunk 22]
```diff
@@ -487,45 +487,53 @@
 
 /proc/pollCultists(var/mob/living/Nominee) // Cult Master Poll
 	var/time_passed = world.time
-	set waitfor = FALSE
 	var/list/yes_voters = new
 	var/list/cult_total = new
-	if(world.time<1800)
+	if(world.time<1)
```
Comment (optimumtact) [inline code/__HELPERS/game.dm]: really jogs the nog

File: code/__HELPERS/game.dm [hunk 23]
```diff
@@ -485,6 +485,68 @@
...
+					cult_total += M
+			if(3)
+				M << "<span class='danger'>Choice registered: Abstain.</span>"
+	sleep(300)
+	if(yes_voters.len > (cult_total.len/2))
+		var/datum/action/innate/cultmast/finalreck/FinalReckoning = new()
+		var/datum/action/innate/cultmast/cultmark/Mark = new()
+		FinalReckoning.Grant(Nominee)
+		Mark.Grant(Nominee)
+		Nominee.mind.special_role = "Cult Master"
```
Comment (ExcessiveUseOfCobblestone) [inline code/__HELPERS/game.dm]: Is there a check to see if this guy gets deconverted/killed during the voting process or is the check purposefully left out for "interesting situations"?

File: code/_globalvars/game_modes.dm [hunk 24]
```diff
@@ -3,3 +3,10 @@ GLOBAL_VAR_INIT(secret_force_mode, "secret") // if this is anything but "secret"
 
 GLOBAL_VAR_INIT(wavesecret, 0) // meteor mode, delays wave progression, terrible name
 GLOBAL_DATUM(start_state, /datum/station_state) // Used in round-end report
+
+// Cult, needs to be global so admin cultists are functional
+GLOBAL_VAR_INIT(blood_target, null) // Cult Master's target or Construct's Master
+GLOBAL_DATUM(sac_mind, /datum/mind)
+GLOBAL_VAR_INIT(sac_image, null)
+GLOBAL_VAR_INIT(cult_mastered, FALSE)
+GLOBAL_VAR_INIT(sac_complete, FALSE)
```
Comment (ExcessiveUseOfCobblestone) [inline code/_globalvars/game_modes.dm]: Can you not just shove these into cult datum?

**Comment by Robustin** [comment]
Well most of Cult mode is broken right now so I'm just gonna have to let the dust settle, very frustrating getting everything working and then having it all blow up because of a refactor out of the blue.

File: code/__HELPERS/game.dm (hunk 23 above)
Comment (Robustin) [inline code/__HELPERS/game.dm]: I figured it was extremely unlikely and not a big issue if it happened. If the cult just elected a guy who can't stay alive 3 minutes into a round then you've got bigger things to worry about. 

But if this would cause runtimes or something  ill add some checks.

File: code/_globalvars/game_modes.dm (hunk 24 above)
Comment (Robustin) [inline code/_globalvars/game_modes.dm]: Simplicity and the encouragement of certain maintainers. They get used quite frequently in several places so I wasn't wild about having to define a cult datum for every single cult proc.

**Comment by Robustin** [comment]
Could you guys pick a different week to suddenly have a tug of war over whether datum antags are in the mind or body?

**Comment by optimumtact** [comment]
thats the secret robustin, everything is constantly changing

**Comment by Robustin** [comment]
Sure, but little refactors that serve no immediate purpose which fuck up big feature PR's that are open should be a basic "no-no".

**Comment by ExcessiveUseOfCobblestone** [comment]
some antags are mind based [traitor] and some are body based [like rev] so we should just have both :^).

File: code/__HELPERS/game.dm [hunk 25]
```diff
@@ -532,16 +530,17 @@
 		Mark.Grant(Nominee)
 		Nominee.mind.special_role = "Cult Master"
 		Nominee.update_action_buttons_icon()
+		Nominee.apply_status_effect(/datum/status_effect/cult_master)
 		SSticker.mode.set_antag_hud(Nominee,"cultmaster")
 		GLOB.cult_mastered = TRUE
 		for(var/datum/mind/B in SSticker.mode.cult)
-			if(isliving(B.current))
-				B.current << "<span class='cultlarge'>[Nominee] has the cult's support and is now their master. Follow their orders to the best of your ability!"
+			if(!B.current.incapacitated())
```
Comment (ChangelingRain) [inline code/__HELPERS/game.dm]: not !, that's a negation; this proc returns TRUE if the target is incapacitated.

File: code/__HELPERS/game.dm [hunk 26]
```diff
@@ -532,16 +530,17 @@
...
-			if(isliving(B.current))
-				B.current << "<span class='cultlarge'>[Nominee] has the cult's support and is now their master. Follow their orders to the best of your ability!"
+			if(!B.current.incapacitated())
+				to_chat(B.current,"<span class='cultlarge'>[Nominee] has the cult's support and is now their master. Follow their orders to the best of your ability!")
 		return TRUE
 	else
 		for(var/datum/mind/B in SSticker.mode.cult)
-			if(isliving(B.current))
-				B.current << "<span class='cultlarge'>[Nominee] could not win the cult's support and shall continue to serve as an acolyte."
+			if(!B.current.incapacitated())
```
Comment (ChangelingRain) [inline code/__HELPERS/game.dm]: here too

File: code/datums/status_effects/buffs.dm [hunk 27]
```diff
@@ -228,3 +228,30 @@
...
+/datum/status_effect/cult_master
+	id = "The Cult Master"
+	duration = -1
+	tick_interval = 100
+	alert_type = null
+	var/alive = TRUE
+
+/datum/status_effect/cult_master/tick()
+	if(!owner)
+		src.Destroy()
```
Comment (ChangelingRain) [inline code/datums/status_effects/buffs.dm]: qdel(src)

File: code/datums/status_effects/buffs.dm [hunk 28]
```diff
@@ -228,3 +228,30 @@
...
+		return
+	if(owner.stat != DEAD && !alive)
+		alive = TRUE
+		return
+	if(owner.stat == DEAD && alive)
+		alive = FALSE
+		on_remove()
+
+/datum/status_effect/cult_master/on_remove()
```
Comment (ChangelingRain) [inline code/datums/status_effects/buffs.dm]: make what you're doing with this proc and call that proc in on_remove(), like below

    /datum/status_effect/cult_master/on_remove()
    remove_message_proc()
    . = ..()

File: code/datums/status_effects/buffs.dm [hunk 29]
```diff
@@ -228,3 +228,30 @@
...
+/datum/status_effect/cult_master/tick()
+	if(!owner)
+		src.Destroy()
+		return
+	if(owner.stat != DEAD && !alive)
+		alive = TRUE
+		return
+	if(owner.stat == DEAD && alive)
+		alive = FALSE
+		on_remove()
```
Comment (ChangelingRain) [inline code/datums/status_effects/buffs.dm]: and when you've done that, call `remove_message_proc()` here instead of on_remove(), because uses of on_remove assume that it'll call . = ..(), which will call qdel(src)

File: code/datums/status_effects/buffs.dm [hunk 30]
```diff
@@ -228,3 +228,30 @@
...
+	tick_interval = 100
+	alert_type = null
+	var/alive = TRUE
+
+/datum/status_effect/cult_master/tick()
+	if(!owner)
+		src.Destroy()
+		return
+	if(owner.stat != DEAD && !alive)
+		alive = TRUE
```
Comment (ChangelingRain) [inline code/datums/status_effects/buffs.dm]: I actually didn't think of doing this, good call here.

**Comment by Robustin** [comment]
Sooooooooooooooooooooooo

**Comment by Sarcalogo** [comment]
\>Giving cult, the most overpowered game mode, even more OP shit
I don't know how it is on /tg/, but Cult wins probably 75% of the time on Hippie. 

\>Infinite melee stun + mutes
\>Free negate 3 ranged stuns shield, doesn't even have to be held in hand
\>Free high force melee weapon
\>Can teleport away instantly using teleport talismans if they are ever in any danger at all (like Security chasing them)
\>Can legally meta game by ghosting at any time
\>If a cultist is arrested, they can just be instantly summoned to be broken out of jail
\>If a cultist is dead, they can be instantly revived with no effort at all
\>If all your cultists are gibbed/incinerated you can just mainfest a shit ton of ghosts
\>You can easily make impenetrable fortresses that NO non cultist can enter unless they have a null rod
\>AOE instacrit 
\>Unstunnable constructs that get hyper speed and an AOE stun, 
\>Then we have juggernauts which are partially immune to lasers as well

I could not think of any way possible to make cult stronger without going to absurd levels like an instant win button. But no.. but cult needs more stuff. L O L. Oldman in charge of game balance. It's gotten to the point where the minute I know it's a cult round I don't even bother fighting them, I just bunker down and spam shuttle call.

**Comment by SgtxRootbeer** [comment]
@Sarcalogo
That's the issue, the shuttle is always the counter to having a cult onboard. It takes time to organize a bunch of idiots that can't handle a simple stunpaper properly, and when the shuttle is the instant red text that it is, cult looses.

**Comment by Robustin** [comment]
@Sarcalogo 

I'm sorry that security sucks that bad on Hippie. I've been tracking stats at https://atlantaned.space/ and Cult is now the most balanced mode that I have ever seen in 7 years of SS13. 

Out of the last 6 months, 4 of those months had EXACTLY that same number of wins and losses. One month was at 50.9% win rate (2 more wins than losses) and last month was 42.1%. 

The strongest cult weapon is a single-use stun item that inflicts ~10 brute on the user. I have seen COUNTLESS robust players tear apart an active cult base by themselves because a single person with a ranged weapon will always beat a bunch of dudes chasing you around with paper. If I had to pick between a base with 8 cultists inside versus the a security officer with a (buckshot) shotgun and a belt of flashbangs or a single atmos tech with an atmos hardsuit and flamethrower, I'd bet against the cultists every single time. 

When we had stat tracking across round length and population, cult's winrate had no significant correlation with population size and it only went down as the round lengthened because after the initial momentum the cult has before its discovered its very easy for security to hold their ground. Once security switches into "shoot anything in maint" mode, it becomes very hard for the cult to keep gaining momentum.

Also bloodboil hasn't been insta-crit for over a year. As if I needed any more of a reason to assume you're just full of shit. 

So TL;DR: Get good and sack up you baby, cult is extremely balanced on our servers - I have actual stats, you have a salty comment.

**Comment by Iamgoofball** [comment]
Hippie has hillariously bad security players, it is a literal case of get good.

**Comment by KorPhaeron** [comment]
But what made cult lose those 50% of rounds? Security evacuating before the cult can do anything? A dumb cult of newbies wandering around and then being shot? A tense, close back and forth between two evenly matched sides? 

What made them win? No security spawned? They made a box in space or lavaland and avoided security the entire game until the "summoning in maint" warning? They repelled a grand raid on their base in a memorable battle?

Winrate is a useful stat but it isn't the end all be all.

I don't think anyone can dispute that the usual reaction of security is to just fucking bail at the sight of the first rune. That doesn't suggest they think it's much fun or very fair to fight the other guys.

I still think having some sort of organization/leader for cult is good but I don't think a 50% winrate automatically means flawless.

**Comment by KorPhaeron** [comment]
Also could you please list what the blood magic actually is in the OP

**Comment by Sarcalogo** [comment]
It doesn't matter how good Security is, even if Security were staffed with identical clones of myself, hell let's say it's staffed with five Dante Smith tier powergamers.  It would make no difference whatsoever. So long as there at least 3 competent cultists the round is already over. The only way to counter cult is to call the shuttle and protect the sacrifice target (or metagame and just incinerate them), you can't actually stamp them out ever. Within ten minutes there are about fifteen cultists on station, and around 1/3 of those have ranged stuns either from Cargo, Medical, or killing a Security Officer. You are fighting magic revs that can:

\>Ghost at any time
They know whatever you are planning and where you are and where you reside. 
\>Teleport away instantly
You can't randomly search people if they manage to dodge the first taser shot you fire, they will just instantly teleport away with a teleport talisman.
\>Teleport other cultists
Let's say you do get someone in cuffs, you can't actually arrest them, if you do they just get instantly teleported away after calling out for help on the radio.
\>Revive themselves from the dead instantly
You have to secure all corpses and guard them, if you don't they'll just wait until you are gone and run off with them. The smarter thing to do would be to just incinerate/gib all cult players you kill. Of course when I did that as Futur I was called a "powergaming shitter who should be banned" so I guess that's not an option :^)))).
\>Make impenetrable bases
They can regroup there if you manage to make a dent in their forces by just outright lasering and incinerating people. They can manifest ghosts and have those ghosts run off murderboning people with free (20 force I think) weapons, and stun shields.

So really your only choice is to just stun, laser, and incinerate anyone who is not Security or get rid of all comms consoles and code red call ASAP. Of course good luck doing that shit as an artificer on crack runs onto screen and spams magic missile while dragging you to cultists who stun talisman you. If you think you are hot shit, come on to Hippie with a hand picked meta crew and all join as Sec, I guarantee you that If I'm a cultist you will lose (unless of course you follow the above powergaming tactics)

It is probably more of a case of /tg/ players being cowards and not actually converting many people than Hippie Security sucking (there are some ridiculous powergaming Security players there that would make Zaross and JLP blush).

**Comment by Sarcalogo** [comment]
In other words you have no arguments and are just pretending as if you do. You also denied my challenge, showing how much of a coward you are, and how right I am.

**Comment by banthebantz** [comment]
WEW LAD
